modules/mssql-server/src/main.tf

```diff
@@ -0,0 +1,56 @@
+resource "azurerm_mssql_server" "main" {
+  name                = "mssql-${var.zone}-${var.environment}-${lookup(local.short_locations, var.location)}-${local.identifier}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  minimum_tls_version          = var.minimum_tls_version
+  version                      = var.sql_server_version
+  administrator_login          = var.azuread_authentication_only ? null : var.administrator_login
+  administrator_login_password = var.azuread_authentication_only ? null : var.administrator_password
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  dynamic "azuread_administrator" {
+    for_each = var.azuread_authentication_only ? [{}] : []
+    content {
+      azuread_authentication_only = var.azuread_authentication_only
+      login_username              = var.azuread_administrator.login_username
+      object_id                   = var.azuread_administrator.object_id
+    }
+  }
+
+  tags = merge(var.tags, local.tags)
+}
+
+resource "azurerm_monitor_diagnostic_setting" "main" {
```

Comment:
Sure the server resource uses Diag Settings? I can't see it either on the resource view in portal or in https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/diagnosticsLogs